common/static/common/js/components/collections/paging_collection.js

```diff
@@ -0,0 +1,218 @@
+ * indexed interface to make consumption easier for views.
+ *
+ * Subclasses may want to override the following properties:
+ *      - url (string): The base url for the API endpoint.
+ *      - isZeroIndexed (boolean): If true, API calls will use page numbers starting at zero. Defaults to false.
+ *      - perPage (number): Count of elements to fetch for each page.
+ *      - server_api (object): Query parameters for the API call. Subclasses may add entries as necessary. By default,
+ *          a 'sort_order' field is included to specify the field to sort on. This field may be removed for subclasses
+ *          that do not support sort ordering, or support it in a non-standard way. By default filterField and
+ *          sortDirection do not affect the API calls. It is up to subclasses to add this information to the appropriate
+ *          query string parameters in server_api.
+ */
+;(function (define) {
+    'use strict';
+    define(['backbone.paginator'], function (BackbonePaginator) {
+        var PagingCollection = BackbonePaginator.requestPager.extend({
+            initialize: function () {
+                // These must be initialized in the constructor because otherwise all PagingCollections would point
+                // to the same object references for sortableFields and filterableFields.
+                this.sortableFields = {};
+                this.filterableFields = {};
+            },
+
+            paginator_core: {
+                type: 'GET',
+                dataType: 'json',
+                url: function () { return this.url; }
+            },
+
+            paginator_ui: {
+                firstPage: function () { return this.isZeroIndexed ? 0 : 1; },
+                currentPage: function () { return this.isZeroIndexed ? 0 : 1; },
+                perPage: function () { return this.perPage; }
+            },
+
+            server_api: {
+                'page': function () { return this.currentPage; },
+                'page_size': function () { return this.perPage; },
+                'sort_order': function () { return this.sortField; }
+            },
+
+            parse: function (response) {
+                this.totalCount = response.count;
+                this.currentPage = response.current_page;
+                this.totalPages = response.num_pages;
+                this.start = response.start;
+                return response.results;
+            },
+
+            isZeroIndexed: false,
+            perPage: 10,
+
+            sortField: '',
+            sortDirection: 'descending',
+            sortableFields: {},
+
+            filterField: '',
+            filterableFields: {},
+
+            /**
+             * Returns the current page number as if numbering starts on page one, regardless of the indexing of the
+             * underlying server API.
+             */
+            currentOneIndexPage: function () {
```

Comment:
I don't like this name. I'd probably just call it ``currentPage``, since this is the external API that is intended to be 1-indexed. Is ``this.currentPage`` something we added, or is it in Backbone.Paginator? If we added it, I'd rename that to indicate that it is the server's current page.